tests/client/client.go

```diff
@@ -6,8 +6,29 @@ import (
 	"fmt"
 	"net/http"
 	"strings"
+	"time"
 )
 
+func Wait(serverURL string, timeout time.Duration) error {
+	testApi := NewTestApiClient(serverURL, &http.Client{})
+
+	// if timeout == 0 {
+	// 	timeout = 24 * time.Hour // This is probably a misuse
+	// }
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	for {
+		if err := testApi.GetEmpty(ctx); err == nil {
+			return nil
+		}
+		if ctx.Err() != nil {
+			return fmt.Errorf("wait: test server still not ready at %v after %v", serverURL, timeout)
+		}
+	}
+}
+
```

Comment:
1. imho, `maxWaitTime` is not necessary, since we get the cancelation from context
2. good idea, I'll make it 100ms
3. nice use of `select {}` pattern :)
4. not necessary, since we're passing ctx to each request anyway

Reply:
1. agree
2. super!
3. if you are serious about it, than thanks , otherwise - i need to learn go even harder :)
4. agree

Reply:
`select {}` used in 8a82cb6b53c14b384c1f165b4d1d0aa142f23d02 ;)